sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/Acl.java

```diff
@@ -18,26 +18,75 @@
 
 import java.lang.annotation.*;
 
+
+/**
+ * Defines ACL configuration for a resource.
+ */
 @Target({ElementType.TYPE, ElementType.METHOD})
 @Retention(RetentionPolicy.RUNTIME)
 @Documented
 public @interface Acl {
 
+  /**
+   * Principals that are allowed to access this resource.
+   * An incoming request must have at least one principal associated with it in this list to be allowed.
+   */
   Matcher[] allow() default {};
 
+
+  /**
+   * After matching an allow rule, an incoming request that has at least one principal
+   * that matches a deny rule will be denied.
+   */
   Matcher[] deny() default {};
 
+
+  /**
+   * The gRPC status code to respond with when access is denied.
+   *
+   * By default, this will be 7 (permission denied), but alternatives might include 16 (unauthenticated) or 5 (not
+   * found). If 0, indicates that the code should be inherited from the parent (regardless of the inherit field).
+   *
+   * When HTTP transcoding is in use, this code will be translated to an equivalent HTTP status code.
+   */
   int denyCode() default 0;
 
+
+  /**
+   * A principal matcher that can be used in an ACL.
+   *
+   * A principal is a very broad concept. It can correlate to a person, a system, or a more abstract concept, such as
+   * the internet.
+   *
+   * A single request may have multiple principals associated with it, for example, it may have come from a particular
+   * source system, and it may have certain credentials associated with it. When a matcher is applied to the request,
+   * the request is considered to match if at least one of the principals attached to the request matches.
+   *
+   * Each Matcher can be configured either with a 'service' or a 'principal', but not both.
```

Comment:
This is an extra line in the docs. In proto, we can use `oneof`, but in java annotations land there is no such a thing. So we need to document. 

We do validate it at runtime.